### PR TITLE
tests: add markdown link checker

### PR DESCRIPTION
**Pull request by andreizet**
**Summary**

This is a build related change, it adds a new action that checks for broken links inside markdown files when performing the CI actions.

It will prevent issues like [#11089](https://github.com/GoogleChrome/lighthouse/pull/11089)

**Related Issues/PRs**

### CODE REVIEW

**Comment by andreizet** [comment]
@googlebot I signed it!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -172,3 +172,14 @@ jobs:
...
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+        with:
+          use-quiet-mode: 'yes'
+          use-verbose-mode: 'yes'
+          config-file: 'markdown.links.config.json'
+          folder-path: 'docs/'
```
Comment (connorjclark) [inline .github/workflows/ci.yml:185]: can this capture the root README.md too?

**Comment by connorjclark** [comment]
Looks like we are blocked on https://github.com/tcort/markdown-link-check/issues/94

**Comment by connorjclark** [comment]
I like their solution: https://github.com/JabRef/jabref/pull/6695

Would be good to move this to a once-a-week check. Could you change the action to be a cron instead?

**Review by andreizet — COMMENTED** [review]
> I like their solution: [JabRef/jabref#6695](https://github.com/JabRef/jabref/pull/6695)
> 
> Would be good to move this to a once-a-week check. Could you change the action to be a cron instead?

Sure, will do.

**Comment by andreizet** [comment]
> Looks like we are blocked on [tcort/markdown-link-check#94](https://github.com/tcort/markdown-link-check/issues/94)

[markdown-link-check](https://github.com/tcort/markdown-link-check) needs an option to retry on 429 codes, but it will not help without this dependency update: [link-check](https://github.com/tcort/link-check/pull/22)

**Comment by andreizet** [comment]
So I've managed to find a solution:
I've splitted the job, so now it checks the files inside /docs, then it checks the ones from root. Also, now it runs every monday at 9 am.

Once the 429 issue will be resolved, I will come back with another pull request, so maybe we can check everything

https://github.com/andreizet/lighthouse/runs/1070765498?check_suite_focus=true

File: docs/readme.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ $ lighthouse --port=9222 --emulated-form-factor=none --throttling.cpuSlowdownMul
 
 ## Lighthouse as trace processor
 
-Lighthouse can be used to analyze trace and performance data collected from other tools (like WebPageTest and ChromeDriver). The `traces` and `devtoolsLogs` artifact items can be provided using a string for the absolute path on disk if they're saved with `.trace.json` and `.devtoolslog.json` file extensions, respectively. The `devtoolsLogs` array is captured from the `Network` and `Page` domains (a la ChromeDriver's [enableNetwork and enablePage options]((https://sites.google.com/a/chromium.org/chromedriver/capabilities#TOC-perfLoggingPrefs-object)).
+Lighthouse can be used to analyze trace and performance data collected from other tools (like WebPageTest and ChromeDriver). The `traces` and `devtoolsLogs` artifact items can be provided using a string for the absolute path on disk if they're saved with `.trace.json` and `.devtoolslog.json` file extensions, respectively. The `devtoolsLogs` array is captured from the `Network` and `Page` domains (a la ChromeDriver's [enableNetwork and enablePage options](https://sites.google.com/a/chromium.org/chromedriver/capabilities#TOC-perfLoggingPrefs-object)).
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply docs/readme.md:136]: nice! this is exactly the kind of thing I wanted to catch.

**Review by connorjclark — APPROVED** [review]
Since this runs so infrequently, and we've seen it handle all the links, we should be good to merge this without any more changes upstream.

Worse case, we get an email some Mondays with a couple "429s"..

Thanks for taking care of this for us!

File: .github/workflows/check-md-links.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+        # checks all markdown files from /docs including all subfolders
+        with:
+          use-quiet-mode: 'yes'
+          use-verbose-mode: 'yes'
+          config-file: 'markdown.links.config.json'
```
Comment (brendankenny) [inline .github/workflows/check-md-links.yml:17]: can this config-file be anywhere? If so, can we move it under `.github/workflows/` to keep it out of the root directory?

Comment (andreizet) [reply .github/workflows/check-md-links.yml]: sure, it's done!

**Review by paulirish — APPROVED** [review]
yeah i also wanted that config file to move out of root. happy that's done. lg!

thanks for working on this @andreizet !

**Pull request merged**